daml-foundations/daml-ghc/ghc-compiler/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -1329,38 +1329,25 @@ isRecordCtor (Ctor _ fldNames fldTys) = not (null fldNames) || null fldTys
 -- then 'maintainer' gets rewritten into
 --
 -- > maintainer = $key._2 ++ $key._1
-rewriteMaintainer :: LF.Expr -> LF.Expr -> ConvertM LF.Expr
-rewriteMaintainer key maintainer = do
-    keyMap <- buildKeyMap key
-    rewriteThisProjections keyMap maintainer
+rewriteMaintainer :: Env -> LF.Expr -> LF.Expr -> ConvertM LF.Expr
```

Comment:
Is this still right for non-complex keys? How can you change the semantics for the old one? Or do you believe this has identical semantics and is just a refactoring?

Reply:
This function still does the right thing for contract keys obeying the syntactic restriction in DAML-LF 1.3. There's a check in the type checker making sure the restriction is obeyed for DAML-LF 1.3.

Reply:
version14 = versionDev for now, then we have a single place to roll. I like that.

Reply:
Done.